### PR TITLE
gadget/quantity: introduce a new package that captures quantities 

### PR DESCRIPTION
**Pull request by bboozzoo**
An idea about capturing gadget related quantities into a single package not fearing circular imports.

### CODE REVIEW

**Review by zyga — COMMENTED** [review]
Looks sensible.

Did you consider other locations, like textutil/quantity or something similar? It's not really specific to the gadget.

File: gadget/quantity/size_test.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+	. "gopkg.in/check.v1"
+	"gopkg.in/yaml.v2"
+
+	"github.com/snapcore/snapd/gadget/quantity"
+)
+
+func TestRun(t *testing.T) { TestingT(t) }
+
+type gadgetSizeTestSuite struct{}
```
Comment (mvo5) [inline gadget/quantity/size_test.go:34]: (nitpick) we probably want to remove gadget from this as it's now pretty neutral about that.

**Review by mvo5 — APPROVED** [review]
I like the idea

**Comment by bboozzoo** [comment]
> Looks sensible.
> 
> Did you consider other locations, like textutil/quantity or something similar? It's not really specific to the gadget.

The parsing is gadget specific though. I tried wrapping the type with another one inside the `gadget` package and hang `UnmarshalYAML` on that type,  but eventually I'd need to introduce additional complexity by having an intermediate gadget structures just for unmarshalling.

**Comment by mvo5** [comment]
Thanks for clarifying this

File: gadget/quantity/size_test.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+		{578 * quantity.SizeMiB, "578 MiB"},
+		{1*quantity.SizeGiB + 123*quantity.SizeMiB, "1.12 GiB"},
+		{1024 * quantity.SizeGiB, "1 TiB"},
+		{2 * 1024 * 1024 * 1024 * quantity.SizeGiB, "2048 PiB"},
+	} {
+		c.Check(tc.size.IECString(), Equals, tc.exp)
+	}
+}
+
+func (s *sizeTestSuite) TestUnmarshalGadgetSize(c *C) {
```
Comment (anonymouse64) [inline gadget/quantity/size_test.go:58]:
```suggestion
func (s *sizeTestSuite) TestUnmarshalYAMLSize(c *C) {
```
?

**Review by anonymouse64 — APPROVED** [review]
Thanks for this! I really like having this separated out so we can use more things with units attached in more places :-)

The test name question is not a blocker, it's fine as is, just wondering if maybe it should be adjusted.

**Pull request merged**